plugins/destination/snowflake/client/client.go

```diff
@@ -19,12 +20,15 @@ type Client struct {
 	logger zerolog.Logger
 	spec   Spec
 	writer *batchwriter.BatchWriter
+
+	setupWriteOnce *sync.Once
```

Comment:
We could use the fancy new https://pkg.go.dev/sync#OnceValue now that we're on Go 1.21

Reply:
We could, but init wasn't a good place for write-related code anyway. Plus if we did that we'd need to use the `ctx` from `New` and not `Write`?

Reply:
Yeah, non blocking comment this is good to go